### ISSUE TITLE
Allow syntax-based ordering for memberSyntaxSortOrder in sort-imports

### ISSUE THREAD

**Issue by briman0094** [eslint/eslint#12065]
**What rule do you want to change?**

`sort-imports`

**Does this change cause the rule to produce more or fewer warnings?**

More

**How will the change be implemented? (New option, new default behavior, etc.)?**

More options for the existing `memberSyntaxSortOrder` option property. Maybe new `ignoreAlphabetical` property?

**Please provide some example code that this change will affect:**

```js
import chalk from "chalk";
import CleanPlugin from "clean-webpack-plugin";
import SassVarsImporter from "./build/lib/SassVarsImporter";
import TsConfigPathsPlugin from "tsconfig-paths-webpack-plugin";

import { BuildInfoPlugin } from "./build/plugins/BuildInfoPlugin";
import { BundleAnalyzerPlugin } from "webpack-bundle-analyzer";
```

**What does the rule currently do for this code?**

My configuration requires that the import order be `"single", "multiple", "all", "none"`. The rule currently suggests that `import { BuildInfoPlugin }` should be sorted alphabetically before `import chalk` because `import { BuildInfoPlugin }` is apparently still considered a "single import" despite having curly braces.

**What will the rule do after it's changed?**

The rule should accept an alternate selection of `memberSyntaxSortOrder` options, those being `"default", "named", "mixed", "all", "none"`. Of course `"all"` and `"none"` would behave like the current options, but instead of classifying the other imports based on the *number of members imported*, it would classify them based on the *syntax style of the import*. This new set of options would be mutually exclusive with the existing set. That is to say, the existing set can be used as-is, but e.g. `"named"` and `"single"` could not be mixed as one relies on import count whereas the other relies on syntax type.

I personally hate having named imports and default imports mixed together without a space in between. I would like to be able to configure the rule to require the following layout, with the options of `"default", "mixed", "named"`:

```js
import Apple from "fruits/apple";

import Banana, { Plantain } from "fruits/banana";

import { Pear } from "fruits";
```

It would also be nice to be able to disable all alphabetization checking, as I don't particularly care about that, but rather keeping the imports organized by syntax.

**Are you willing to submit a pull request to implement this change?**

I don't think I'll have time, but if it sits stagnant without someone picking it up for a while, I can try and find time for it.

**Comment by mdjermanovic** [comment]
Hi @briman0094, sorry for waiting this long.

I'm :thumbsup: for this in general, and the solution you suggested looks good to me, as it is a non-breaking change.

There are certain design questions, I'm not sure is there a similar example in other rules. In particular, can the schema validation support this.
